# Automatically generated by Pynguin.
import pytest
import typ.test_case as module_0
import inspect as module_1
import dataclasses as module_2

@pytest.mark.xfail(strict=True)
def test_case_0():
    main_test_case_0 = module_0.MainTestCase()
    main_test_case_0.assert_files(main_test_case_0, main_test_case_0, main_test_case_0)

@pytest.mark.xfail(strict=True)
def test_case_1():
    main_test_case_0 = module_0.MainTestCase()
    var_0 = module_1.get_annotations(main_test_case_0)
    var_1 = main_test_case_0.assert_files(var_0, var_0, var_0)
    var_1.check(main_test_case_0, var_0, aenv=var_0, ret=var_1, out=var_0, err=main_test_case_0, rerr=var_0, exp_files=main_test_case_0, files_to_ignore=var_0, universal_newlines=var_0)

@pytest.mark.xfail(strict=True)
def test_case_2():
    main_test_case_0 = module_0.MainTestCase()
    main_test_case_0.check(files=main_test_case_0, prog=main_test_case_0, err=main_test_case_0, universal_newlines=main_test_case_0)

@pytest.mark.xfail(strict=True)
def test_case_3():
    main_test_case_0 = module_0.MainTestCase()
    main_test_case_0.check(stdin=main_test_case_0, files=main_test_case_0, cwd=main_test_case_0, ret=main_test_case_0, out=main_test_case_0, rerr=main_test_case_0, exp_files=main_test_case_0)

def test_case_4():
    test_case_0 = module_0.TestCase()
    assert module_0.TestCase.maxDiff == 5280

@pytest.mark.xfail(strict=True)
def test_case_5():
    none_type_0 = None
    module_0.convert_newlines(none_type_0)

@pytest.mark.xfail(strict=True)
def test_case_6():
    main_test_case_0 = module_0.MainTestCase()
    var_0 = main_test_case_0.assertEqual(main_test_case_0, main_test_case_0)
    main_test_case_1 = module_0.MainTestCase()
    main_test_case_1.call(main_test_case_1, main_test_case_1, main_test_case_1, main_test_case_1)

@pytest.mark.xfail(strict=True)
def test_case_7():
    main_test_case_0 = module_0.MainTestCase()
    main_test_case_0.check(aenv=main_test_case_0, host=main_test_case_0)

@pytest.mark.xfail(strict=True)
def test_case_8():
    main_test_case_0 = module_0.MainTestCase()
    main_test_case_0.check(main_test_case_0, main_test_case_0, main_test_case_0, files=main_test_case_0, host=main_test_case_0, universal_newlines=main_test_case_0)

@pytest.mark.xfail(strict=True)
def test_case_9():
    main_test_case_0 = module_0.MainTestCase()
    var_0 = module_2.is_dataclass(main_test_case_0)
    var_1 = var_0.__repr__()
    var_2 = var_0.__eq__(main_test_case_0)
    main_test_case_0.check(var_1, env=var_2, files=var_0, host=main_test_case_0, err=var_1, rerr=var_1)

@pytest.mark.xfail(strict=True)
def test_case_10():
    main_test_case_0 = module_0.MainTestCase()
    var_0 = module_1.get_annotations(main_test_case_0)
    var_1 = var_0.setdefault(main_test_case_0)
    main_test_case_0.assert_files(var_0, var_0, var_0)